components/UI/Drawer/Drawer.tsx

```diff
@@ -82,6 +99,32 @@ const Drawer = () => {
   );
   const [showSavedData, setShowSavedData] = useState(true);
 
+  const { data, isLoading, error } = useSWR<Data | undefined>(
+    locationsURL(drawerData?.reference),
+    dataFetcher
+  );
+
+  useEffect(() => {
+    if (isOpen) {
```

Comment:
Butun fonksiyonu if ile sarmalamak yerine useEffect'in basinda `if (!isOpen) return;` diyerek okunabilirligi arttirabilirsin.